app/serializers/user_bookmark_base_serializer.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class UserBookmarkBaseSerializer < ApplicationSerializer
+  attributes :id,
+             :created_at,
+             :updated_at,
+             :name,
+             :reminder_at,
+             :pinned,
+             :title,
+             :fancy_title,
+             :excerpt,
+             :bookmarkable_id,
+             :bookmarkable_type,
+             :bookmarkable_url
+
+  def title
+    raise NotImplementedError
+  end
+
+  def fancy_title
+    raise NotImplementedError
+  end
+
+  def cooked
+    raise NotImplementedError
+  end
+
+  def bookmarkable_url
+    raise NotImplementedError
+  end
+
+  def excerpt
+    raise NotImplementedError
+  end
+
+  has_one :user, serializer: BasicUserSerializer, embed: :objects
```

Comment:
Just curious, what if the item being bookmarked does not have a relation to user?

Reply:
This is the user attached to the bookmark, not the bookmarkable thing.

Reply:
Actually no sorry you are right. I think we will cross that bridge when we come to it, will add a note for now. I don't think it's a big deal, just is used to show the avatar in the bookmark list.